src/test/scala/chiselTests/Counter.scala

```diff
@@ -38,6 +40,21 @@ class WrapTester(max: Int) extends BasicTester {
   }
 }
 
+class InitTester(max: Int, init: Int) extends BasicTester {
+  val cntFromInit = Counter(max, init)
+  cntFromInit.inc()
+  
+  val cntFromZero = Counter(max)
+  val wrap = cntFromZero.inc()
+
+  val restart = if (init > 0) cntFromZero.value === (init - 1).U else false.B
+  when (restart) { cntFromInit.restart() }
+
+  assert(! (cntFromZero.value >  init.U) || (cntFromInit.value === cntFromZero.value) )
```

Comment:
This seems hard to understand. Can you just negate the comparison instead of adding an external negation? And what's with the weird spacing?